test/jest-light-runner/src/worker-runner.js

```diff
@@ -0,0 +1,207 @@
+) {
+  await runHooks("beforeAll", block, results, stats, ancestors);
+
+  for (const child of block.children) {
+    const { type, mode, fn, name } = child;
+    const nextAncestors = ancestors.concat(name);
+
+    if (
+      mode === "skip" ||
+      (hasFocusedTests && type === "test" && mode !== "only")
+    ) {
+      stats.pending++;
+      results.push({ ancestors, title: name, errors: [], skipped: true });
+    } else if (type === "describeBlock") {
+      await runTestBlock(child, hasFocusedTests, results, stats, nextAncestors);
+    } else if (type === "test") {
+      await runHooks("beforeEach", block, results, stats, nextAncestors, true);
+      await runTest(fn, stats, results, ancestors, name);
+      await runHooks("afterEach", block, results, stats, nextAncestors, true);
+    }
+  }
+
+  await runHooks("afterAll", block, results, stats, ancestors);
+
+  return results;
+}
+
+async function runTest(fn, stats, results, ancestors, name) {
+  expect.setState({
+    suppressedErrors: [],
+    currentTestName: ancestors.concat(name).join(" "),
+  });
+
+  const errors = [];
+  await callAsync(fn).catch(error => {
+    errors.push(error);
+  });
+
+  // Get suppressed errors from ``jest-matchers`` that weren't throw during
+  // test execution and add them to the test result, potentially failing
+  // a passing test.
+  const { suppressedErrors } = expect.getState();
+  expect.setState({ suppressedErrors: [] });
+  if (suppressedErrors.length > 0) {
+    errors.unshift(...suppressedErrors);
+  }
+
+  if (errors.length > 0) {
+    stats.failures++;
+  } else {
+    stats.passes++;
+  }
+  results.push({ ancestors, title: name, errors, skipped: false });
+}
+
+async function runHooks(hook, block, results, stats, ancestors, runInParents) {
+  for (const { type, fn } of block.hooks) {
+    if (type === hook) {
+      await callAsync(fn).catch(error => {
+        stats.failures++;
+        results.push({ ancestors, title: `(${hook})`, error, skipped: false });
+      });
+    }
+  }
+
+  if (block.parent && runInParents) {
+    await runHooks(hook, block.parent, results, stats, ancestors, true);
+  }
+}
+
+function callAsync(fn) {
+  if (fn.length >= 1) {
+    return new Promise((resolve, reject) => {
+      fn((err, result) => {
+        if (err) {
+          reject(err);
+        } else {
+          resolve(result);
+        }
+      });
+    });
+  } else {
+    return Promise.resolve().then(fn);
+  }
+}
+
+/**
+ *
+ * @param {*} stats
+ * @param {*} tests
+ * @param {*} path
+ * @returns {import("@jest/test-result").TestResult}
+ */
+function toTestResult(stats, tests, path) {
+  return {
+    coverage: globalThis.__coverage__,
+    console: null,
+    failureMessage: tests
+      .filter(t => t.errors.length > 0)
+      .map(failureToString)
+      .join("\n"),
+    numFailingTests: stats.failures,
+    numPassingTests: stats.passes,
+    numPendingTests: stats.pending,
+    perfStats: {
+      end: +new Date(stats.end),
```

Comment:
`end` and `start` seem to be not yet implemented. They are always `undefined` on my local machine.

We can implement using the performance API available since Node 8.5.

The `perfStats` also includes `slow` and `duration`, which can be accordingly implemented

https://github.com/facebook/jest/blob/3a85065fe5604655e1337ffc1631f9999722c821/packages/jest-runner/src/runTest.ts#L293-L299